openshift/dynamic/client.py

```diff
@@ -657,13 +657,16 @@ def parse_api_groups(self, request_resources=False):
                 new_group = {}
                 for version_raw in group['versions']:
                     version = version_raw['version']
+                    resource_group = self._cache.get('resources', {}).get(DISCOVERY_PREFIX, {}).get(group['name'], {}).get(version)
                     preferred = version_raw == group['preferredVersion']
-                    resources = {}
+                    resources = resource_group.resources if resource_group else {}
                     if request_resources:
                         resources = self.get_resources_for_api_version(DISCOVERY_PREFIX, group['name'], version, preferred)
                     new_group[version] = ResourceGroup(preferred, resources=resources)
                 groups[DISCOVERY_PREFIX][group['name']] = new_group
-            self._cache['resources'] = groups
+                self._cache['resources'].update(groups)
```

Comment:
Wouldn't doing the dictionary update every iteration of the loop be slower than just setting self._cache['resources'] at the end?

Reply:
woops, typo, that was supposed to be one level out